packages/nextjs/app/page.tsx

```diff
@@ -1,52 +1,247 @@
 "use client";
 
+import { useEffect, useState } from "react";
+import Image from "next/image";
 import Link from "next/link";
 import type { NextPage } from "next";
 import { BugAntIcon, MagnifyingGlassIcon } from "@heroicons/react/24/outline";
 
-const Home: NextPage = () => {
+const images: string[] = []; // TODO: add the images of batch members
+
+interface SquareProps {
```

Comment:
There is too much code in the same file. Let's separate each component into its own file.